rust/kernel/sync/arc.rs

```diff
@@ -7,178 +7,216 @@
 //! underlying object when it reaches zero. It is also safe to use concurrently from multiple
 //! threads.
 //!
-//! It is different from the standard library's [`Arc`] in two ways: it does not support weak
-//! references, which allows it to be smaller -- a single pointer-sized integer; it allows users to
-//! safely increment the reference count from a single reference to the underlying object.
+//! It is different from the standard library's [`Arc`] in a few ways:
+//! 1. It is backed by the kernel's `refcount_t` type.
+//! 2. It does not support weak references, which allows it to be half the size.
+//! 3. It saturates the reference count instead of aborting when it goes over a threshold.
+//! 4. It does not provide a `get_mut` method, so the ref counted object is pinned.
```

Comment:
There is no way to get a mutable reference to the `T` from a `Ref<T>`. It doesn't matter if `T: Unpin`.

Here I'm using the _pin_ meaning as in Rust, from https://doc.rust-lang.org/std/pin/index.html:

> At a high level, a `Pin<P>` ensures that the pointee of any pointer type P has a stable location in memory, meaning it cannot be moved elsewhere and its memory cannot be deallocated until it gets dropped. We say that the pointee is “pinned”. 

I suppose I can copy and paste the wording from above, but I'm not sure if this is better than saying 'pinned' -- it may give readers the impression that this has a [potentially subtle] different meaning than the regular one in Rust.

Reply:
Yes, I agree. Especially now we've established that `try_new_and_init()` exposes a `Pin<&mut T>`, the word "pinned" is probably fully correct here.